core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -730,20 +733,46 @@ public AvroCompatibilityLevel getCompatibilityLevel(String subject)
   @Override
   public boolean isCompatible(String subject,
                               String newSchemaObj,
-                              String latestSchema)
+                              String latestSchema) 
       throws SchemaRegistryException {
-    AvroSchema newAvroSchema = AvroUtils.parseSchema(newSchemaObj);
-    AvroSchema latestAvroSchema = AvroUtils.parseSchema(latestSchema);
     if (latestSchema == null) {
       throw new InvalidSchemaException(
-          "Existing schema " + latestSchema + " is not a valid Avro schema");
+        "Latest schema not provided");
```

Comment:
It looks like this condition may not even be possible anymore? This method looks like it might only be invoked from one location where the last schema cannot be `null`. Shouldn't really be an issue if we have the extra check, just not sure it's relevant anymore.